sources/packages/backend/apps/queue-consumers/src/processors/schedulers/workflow/_tests_/assessment-workflow-enqueuer.scheduler.e2e-spec.ts

```diff
@@ -186,6 +195,78 @@ describe(
       expect(startApplicationAssessmentQueueMock.add).not.toBeCalled();
     });
 
+    it(`Should queue the assessment when an application has one assessment with status '${StudentAssessmentStatus.CancellationRequested}'.`, async () => {
+      // Arrange
+
+      // Application submitted with original assessment and completed assessment.
+      const application = await createDefaultApplication(
+        StudentAssessmentStatus.CancellationRequested,
+      );
+      application.applicationStatus = ApplicationStatus.Overwritten;
+      // Force currentAssessment and currentProcessingAssessment to be the same to test the SQL condition.
```

Comment:
I believe that there is no need to have this data setup. It is supposed to matter only to start an assessment.
Please let me know if I am missing something.

Reply:
Yes, you're right. I'm just keeping the `currentAssessment` assignment to be easier to assert it.